### PR TITLE
reusable pagination for level 2 components

### PR DESCRIPTION
**Pull request by madvincy**

### CODE REVIEW

File: src/summary/history/pagination.test.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    });
+    await wait(() => {
+      const nextBtn = getByText(/next/i);
+      expect(nextBtn).not.toBeNull;
+      fireEvent.click(nextBtn);
+      expect(nextBtn).toBeNull;
+      expect(getByText(/previous/i)).not.toBeNull;
+    });
+  });
+  it("should not display next on the last page", async () => {
```
Comment (fatmali) [inline src/summary/history/pagination.test.tsx]: repetition of test description.

File: src/summary/history/pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      <tbody>
+        <tr
+          style={{
+            backgroundColor: "var(--omrs-color-bg-medium-contrast)",
+            color: "var(--omrs-color-ink-medium-contrast)",
+            padding: "0.625rem 1.625rem 0.625rem 1rem"
+          }}
+        >
+          {showPreviousButton && (
+            <td className={styles.moreNotes}>
```
Comment (fatmali) [inline src/summary/history/pagination.tsx]: rename this css class to be less specific to notes.

File: src/summary/history/pagination.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      : setShowPreviousButton(false);
+    paginate();
+  }, [totalData, currentPage, resultsPerPage]);
+  useEffect(() => {
+    setTotalData(props.totalData);
+    setResultsPerPage(props.resultsPerPage);
+    setPageNumbers(Math.ceil(props.totalData / props.resultsPerPage));
+  }, [props]);
+  return (
+    <table style={{ width: "100%" }}>
```
Comment (fatmali) [inline src/summary/history/pagination.tsx]: why are you using a table here?

File: src/summary/history/pagination.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    indexOfFirstResult = indexOfLastResult;
+    indexOfLastResult = indexOfFirstResult + resultsPerPage;
+    setCurrentPage(currentPage + 1);
+  }
+  function previousPage() {
+    indexOfFirstResult = indexOfFirstResult - resultsPerPage;
+    indexOfLastResult = indexOfFirstResult + resultsPerPage;
+    setCurrentPage(currentPage - 1);
+  }
+  function paginate() {
```
Comment (fatmali) [inline src/summary/history/pagination.tsx]: can you describe what this function aims to achieve?

Comment (madvincy) [reply src/summary/history/pagination.tsx]: most of the level two components have been made using tables,

File: src/summary/history/pagination.tsx (hunk 3 above)
Comment (madvincy) [reply src/summary/history/pagination.tsx]: most of the level two components have been made using tables,

File: src/summary/history/pagination.tsx (hunk 4 above)
Comment (jonathandick) [reply src/summary/history/pagination.tsx]: please rename to pagination.component.tsx

**Pull request closed without merging**